### PR TITLE
[OpenTURNS] add new port

### PR DESCRIPTION
**Pull request by Neumann-A**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies ... controlled by USE_ variables which are all ON.
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [?] The generated "usage text" is accurate. See docs/examples/adding-an-explicit-usage.md for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by Neumann-A** [comment]
Ok static stuff seems to be just missing linking gmp

File: scripts/ci.baseline.txt [hunk 1]
```diff
@@ -732,6 +732,7 @@ openscap:x64-windows-static=fail
 openscap:x64-osx=fail
 # https://github.com/AcademySoftwareFoundation/openvdb/issues/1362
 # openvdb\openvdb\libopenvdb.lib : fatal error LNK1248: image size (109A36020) exceeds maximum allowable size (FFFFFFFF)
+openturns:x64-osx=fail
```
Comment (BillyONeal) [inline scripts/ci.baseline.txt:735]: This needs to be above the above comment talking about openvdb.

**Review by BillyONeal — COMMENTED** [review]
> @BillyONeal any hint how i should solve the osx configure error here? Seems like the bison version on the vm is too low. Only has 2.3 and 2.4 is required.

I think a baseline skip is appropriate for that.

> This is one of the reasons why I think we need tool ports to deal with the versions of those in the same way as the rest.

Tool ports come with their own problems though. For instance, they expose the tool to the installed tree which is generally undesirable for toolset parts. And they effectively never take things from the system even if those are perfectly good. (They make the `VCPKG_FORCE_SYSTEM_BINARIES` / we fail on arm linux or alpine etc. problem worse) I'm on the hook to have a cohesive answer for fetch/tool-ports/artifacts/find_acquire_program soon, will keep you posted.

File: scripts/ci.baseline.txt (hunk 1 above)
Comment (Neumann-A) [inline scripts/ci.baseline.txt:735]:
```suggestion
openturns:x64-osx=fail
# https://github.com/AcademySoftwareFoundation/openvdb/issues/1362
# openvdb\openvdb\libopenvdb.lib : fatal error LNK1248: image size (109A36020) exceeds maximum allowable size (FFFFFFFF)
```

**Review by BillyONeal — DISMISSED** [review]
This LGTM as long as the usage makes sense to you, @Neumann-A , can you confirm?

Thanks for the new port!

**Comment by Neumann-A** [comment]
> can you confirm?

I mean it is the same usage as the heuristics and it looked right to me considering what targets gets expored which is why i didn't bother to add a usagge.

**Comment by BillyONeal** [comment]
I've been adding them just to get rid of the heuristic part of the message and asking folks who are adding ports to confirm that's what they expect. (When a new port is added hopefully the person adding knows what they wanted to do with it?)

**Comment by dg0yt** [comment]
> When a new port is added hopefully the person adding knows what they wanted to do with it?

This may be to optimistic. People may have their own find procedures or do not use CMake at all. And even our reviewers didn't object against obviously false results (`find_package(cmake REQUIURED)`). However, my changes seem to have eliminated a big share of the bad output, in related to subdirectories and find modules.

**Comment by Neumann-A** [comment]
I just need the port to fix the vtk all feature. As far as I can tell usage there is via variables and not targets.

**Comment by Neumann-A** [comment]
@MonicaLiu0311 please remove the tag author-response (at least I don't know what to response to here.)

File: ports/openturns/vcpkg.json [hunk 2]
```diff
@@ -0,0 +1,46 @@
+{
+  "name": "openturns",
+  "version": "1.2.0",
```
Comment (MonicaLiu0311) [inline ports/openturns/vcpkg.json:3]: I didn't find `1.2.0`, the latest version is [1.20](https://github.com/openturns/openturns/releases/tag/v1.20).

**Comment by MonicaLiu0311** [comment]
- [x] The generated "usage text" is accurate.

```
openturns provides CMake targets:

    # this is heuristically generated, and may not be correct
    find_package(OpenTURNS CONFIG REQUIRED)
    target_link_libraries(main PRIVATE OT)
```
The usage test passed(The header files can be found.).

**Review by MonicaLiu0311 — APPROVED** [review]

**Pull request merged**